pkg/cmd/cli/cmd/set/volume.go

```diff
@@ -532,6 +549,8 @@ func (v *VolumeOptions) getVolumeUpdatePatches(infos []*resource.Info, singular
 }
 
 func setVolumeSourceByType(kv *kapi.Volume, opts *AddVolumeOptions) error {
+	defaultMode, _ := strconv.ParseInt(opts.DefaultMode, 8, 32)
```

Comment:
If we keep the check in Validate, we can be sure that DefaultMode is valid here. If we do not keep the check, I agree. However, perhaps we should use 0644 as the backup, since that would be more expected by the user?